students/dave_arasim/session03/list_lab.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+'''List Lab exercise for Session 3, Python 210
+Written by David K. Arasim - 10/17/18'''
+
+print('*'*20, ' Series 1 requirements ', '*'*20)
+print()
+
+fruit_list = ['Apples', 'Pears', 'Oranges', 'Peaches']
+
+print(fruit_list)
+print()
+
+print('Please enter another fruit to add to the list: ', end='')
+new_fruit = input()
+print()
+
+if new_fruit != '':
+    fruit_list.append(new_fruit)
+
+print(fruit_list)
+print()
+
+fruit_list_ct = len(fruit_list)
+
+print('Please enter the fruit number to display (1 - ', fruit_list_ct, '): ', sep='', end='')
+fruit_num = input()
+try:
+    fruit_num = int(fruit_num)
+except ValueError:
+    fruit_num = ''
+
+if fruit_num == '':
+    print('Sorry, but that is not a number')
+else:
+    if (fruit_num >= 1) and (fruit_num <= fruit_list_ct):
+        print(fruit_num, ': ', fruit_list[(fruit_num-1)], sep='')
+    else:
+        print('That fruit number: ', fruit_num, ' is out of range', sep='')
+print()
+
+print(fruit_list)
+print()
+
+print('Please enter another fruit to insert at the beginning of the list: ', end='')
+new_fruit = input()
+print()
+
+if new_fruit != '':
+    fruit_list = [new_fruit] + fruit_list  #Inserting using '+'
+
+print(fruit_list)
+print()
+
+print('Please enter another fruit to insert at the beginning of the list: ', end='')
+new_fruit = input()
+print()
+
+if new_fruit != '':
+    fruit_list.insert(0, new_fruit)      #Inserting using insert()
+
+print(fruit_list)
+print()
+
+search_letter = 'P'
+
+print('Here are the fruits that begin with "', search_letter, '": ', sep='')
+for this_fruit in fruit_list:
+    if this_fruit[0] == search_letter:
+        print(this_fruit, end=' ')
+print()
+print()
+
+###################################################################################################
+
+print('*'*20, ' Series 2 requirements ', '*'*20)
+print()
+
+print(fruit_list)
+print()
+
+fruit_list.pop()
+
+print('Removing last fruit in list with pop():')
+print(fruit_list)
+print()
+
+print('Fruit to delete: ', end='')
+fruit_name = input()
+print()
+
+if fruit_name == '':
+    print('Nothing input...delete cancelled.')
+else:
+    fruit_found = False
+    for i,this_fruit in reversed(list(enumerate(fruit_list))):
```

Comment:
Hmm... I'd say it's an interesting way of solving the `.pop()` indexing issue, but I suppose you could use `fruit_list.remove(fruit_name)`? Of course, it should first satisfy the problem requirements (which are vague to me at this time... sorry), but I don't think the intention of this problem is to apply a nontrivial workaround like this... Just a thought and comment. No need to fix this code.